### PR TITLE
suite-native: Blur phishing txns

### LINKED ISSUE #13241: Polygon - blur suspicious transactions
Blur phishing txns

### PR DESCRIPTION
**Pull request by vytick**
- fixed bad rounded corners for eth token txns
- blur phishing txns in the list using DiscreetText

Visual approved by @sime 

For testing you can [comment this row](https://github.com/trezor/trezor-suite/blob/ffaf7286b3b433452fb6480b46d7405d07b6463b/suite-native/tokens/src/ethereumTokensSelectors.ts#L137) and use all seed `Ethereum #1` account, in detail select `Include tokens` and there will be one phishing txn. According to discussion with @sime we keep current setup for visibility of txn so no change in filtering was done.

## Related Issue

Resolve #13241

### CODE REVIEW

File: suite-native/formatters/src/components/FiatAmountFormatter.tsx [hunk 1]
```diff
@@ -14,10 +14,17 @@ type FiatAmountFormatterProps = FormatterProps<string | null> &
...
         isDiscreetText?: boolean;
+        isForcedDiscreetMode?: boolean;
     };
 
 export const FiatAmountFormatter = React.memo(
-    ({ network, value, isDiscreetText = true, ...textProps }: FiatAmountFormatterProps) => {
+    ({
+        network,
+        value,
+        isDiscreetText = true,
```
Comment (PeKne) [inline suite-native/formatters/src/components/FiatAmountFormatter.tsx]: nit: Put the prop with default value to the end please.

File: suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx [hunk 2]
```diff
@@ -43,13 +53,16 @@ export const TransactionListItemValues = ({
                 <EmptyAmountText />
             ) : (
                 <Box flexDirection="row">
-                    <SignValueFormatter value={getTransactionValueSign(transaction.type)} />
+                    {!isPhishingTransaction && (
```
Comment (PeKne) [inline suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx]: It is quite strange to me that the sign is hidden even though the transaction label still says `Sent` and the icon is arrow up for send transactions. Similar for other types.

File: suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx [hunk 3]
```diff
@@ -72,31 +86,27 @@ export const TransactionListItem = ({
...
-        return (
-            <TokenTransferListItem
-                transaction={transaction}
-                accountKey={accountKey}
-                txid={transaction.txid}
-                tokenTransfer={transaction.tokens[0]}
-                includedCoinsCount={transaction.tokens.length - 1}
-            />
-        );
+    const symbolOrTokenTransfer = isTokenOnlyTransaction
```
Comment (PeKne) [inline suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx]: why is this needed? IMO if you pass both tokenTransfer and networkSymbol is should behave the same. Seems overcomplicated to me.

File: suite-native/transactions/src/components/TransactionsList/TransactionListItemContainer.tsx [hunk 4]
```diff
@@ -166,9 +166,13 @@ export const TransactionListItemContainer = ({
     );
 
     const iconColor: Color = isTransactionPending ? 'backgroundAlertYellowBold' : 'iconSubdued';
-    const coinSymbol = tokenTransfer?.contract ?? networkSymbol;
+    const coinSymbol = !isPhishingTransaction
```
Comment (PeKne) [inline suite-native/transactions/src/components/TransactionsList/TransactionListItemContainer.tsx]: nit: Not negated condtionals are better readable. Please change it to `isPhishingTransaction ? B : A`

File: suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx [hunk 5]
```diff
@@ -72,31 +86,27 @@ export const TransactionListItem = ({
     isFirst = false,
     isLast = false,
 }: TransactionListItemProps) => {
-    const includedCoinsCount = areTokensIncluded ? transaction.tokens.length : 0;
-
     const isTokenOnlyTransaction =
         areTokensIncluded && transaction.amount === '0' && transaction.tokens.length !== 0;
```
Comment (PeKne) [inline suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx]: I don't understand why wy can not use TokenTranferListItem anymore? Please clarify.

File: suite-native/formatters/src/components/CryptoToFiatAmountFormatter.tsx [hunk 6]
```diff
@@ -19,6 +20,7 @@ export const CryptoToFiatAmountFormatter = ({
     historicRate,
     useHistoricRate,
     isBalance = false,
+    isForcedDiscreetMode,
```
Comment (PeKne) [inline suite-native/formatters/src/components/CryptoToFiatAmountFormatter.tsx]:
```suggestion
```

File: suite-native/formatters/src/components/CryptoToFiatAmountFormatter.tsx [hunk 7]
```diff
@@ -29,5 +31,12 @@ export const CryptoToFiatAmountFormatter = ({
         cryptoValue: value,
     });
 
-    return <FiatAmountFormatter network={network} value={fiatValue} {...textProps} />;
+    return (
+        <FiatAmountFormatter
+            network={network}
+            value={fiatValue}
+            isForcedDiscreetMode={isForcedDiscreetMode}
```
Comment (PeKne) [inline suite-native/formatters/src/components/CryptoToFiatAmountFormatter.tsx]:
```suggestion
```

the `isForcedDiscreetMode` prop will be passed automatically via the `...textProps`

File: suite-native/formatters/src/components/EthereumTokenAmountFormatter.tsx [hunk 8]
```diff
@@ -20,6 +21,7 @@ export const EthereumTokenAmountFormatter = ({
     decimals = 0,
     variant = 'hint',
     color = 'textSubdued',
+    isForcedDiscreetMode,
```
Comment (PeKne) [inline suite-native/formatters/src/components/EthereumTokenAmountFormatter.tsx]: same here

File: suite-native/formatters/src/components/EthereumTokenToFiatAmountFormatter.tsx [hunk 9]
```diff
@@ -28,6 +29,7 @@ export const EthereumTokenToFiatAmountFormatter = ({
     numberOfLines,
     historicRate,
     useHistoricRate,
+    isForcedDiscreetMode,
```
Comment (PeKne) [inline suite-native/formatters/src/components/EthereumTokenToFiatAmountFormatter.tsx]: same here

File: suite-native/formatters/src/components/FiatAmountFormatter.tsx [hunk 10]
```diff
@@ -14,10 +14,17 @@ type FiatAmountFormatterProps = FormatterProps<string | null> &
     TextProps & {
         network?: NetworkSymbol;
         isDiscreetText?: boolean;
+        isForcedDiscreetMode?: boolean;
     };
 
 export const FiatAmountFormatter = React.memo(
-    ({ network, value, isDiscreetText = true, ...textProps }: FiatAmountFormatterProps) => {
+    ({
```
Comment (PeKne) [inline suite-native/formatters/src/components/FiatAmountFormatter.tsx]: same

**Review by PeKne — COMMENTED** [review]
I have some questions here. Otherwise, I tested and it worked well.

File: suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx (hunk 3 above)
Comment (vytick) [reply suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx]: `TransactionListItemContainerProps` `RequireExactlyOne` of those two and differentiates based on this

File: suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx (hunk 5 above)
Comment (vytick) [reply suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx]: Good point, I was confused by almost the same name of component wrapped in the TransactionListItemContainer inside TokenTranferListItem. I'll change it back and update 👍

File: suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx (hunk 3 above)
Comment (vytick) [reply suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx]: but I will update this with TokenTransferListItem

Comment (vytick) [reply suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx]: 9e32cee8e9

File: suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx (hunk 5 above)
Comment (vytick) [reply suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx]: 9e32cee8e9

File: suite-native/formatters/src/components/FiatAmountFormatter.tsx (hunk 1 above)
Comment (vytick) [reply suite-native/formatters/src/components/FiatAmountFormatter.tsx]: 4a845d4e24

File: suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx (hunk 2 above)
Comment (vytick) [reply suite-native/transactions/src/components/TransactionsList/TransactionListItem.tsx]: 4a845d4e24

File: suite-native/transactions/src/components/TransactionsList/TransactionListItemContainer.tsx (hunk 4 above)
Comment (vytick) [reply suite-native/transactions/src/components/TransactionsList/TransactionListItemContainer.tsx]: 4a845d4e24

File: suite-native/formatters/src/components/CryptoToFiatAmountFormatter.tsx (hunk 6 above)
Comment (vytick) [reply suite-native/formatters/src/components/CryptoToFiatAmountFormatter.tsx]: 4a845d4e24

File: suite-native/formatters/src/components/CryptoToFiatAmountFormatter.tsx (hunk 7 above)
Comment (vytick) [reply suite-native/formatters/src/components/CryptoToFiatAmountFormatter.tsx]: good point! 4a845d4e24

File: suite-native/formatters/src/components/EthereumTokenAmountFormatter.tsx (hunk 8 above)
Comment (vytick) [reply suite-native/formatters/src/components/EthereumTokenAmountFormatter.tsx]: 4a845d4e24

File: suite-native/formatters/src/components/EthereumTokenToFiatAmountFormatter.tsx (hunk 9 above)
Comment (vytick) [reply suite-native/formatters/src/components/EthereumTokenToFiatAmountFormatter.tsx]: 4a845d4e24

File: suite-native/formatters/src/components/FiatAmountFormatter.tsx (hunk 10 above)
Comment (vytick) [reply suite-native/formatters/src/components/FiatAmountFormatter.tsx]: 4a845d4e24

**Review by Nodonisko — APPROVED** [review]

**Review by PeKne — APPROVED** [review]
Thanks for addressing my comments. 💪
